docs/reference/runtimes/python/python-reference.md

```diff
@@ -64,6 +66,40 @@ COPY . /opt/nuclio
 CMD [ "processor" ]
 ```
 
+<a id="27-eol"></a>
+## Python runtime 2.7 EOL
+
+Official Python 2.7 is no longer maintained, and it has reached its Nuclio End of life due date.
+That means, starting from Nuclio >= 1.6.0 you would  not be able to deploy any Nuclio function using Python 2.7 runtime.
+
+To reach better throughput and high performance, I suggest reading [Introducing Python runtime 3.7 and 3.8](#introducing-python-runtime-37-and-38), 
+and migrate your function to Python 3.8
+
+
+<a id="introducing-37-and-38"></a>
+## Introducing Python runtime 3.7 and 3.8
+
+Nuclio is now officially support python 3.7 and python 3.8 along with good-old python 3.6.
+
+Key changes and differences:
+
+- Python 3.7 and 3.8 base images are `python:3.7` and `python:3.8`, respectively.
+- Events metadata, such as headers, path, method, etc are now byte-string.
+  e.g.: if your code used event metadata, such as `event.path` -
+
+    ```python
+    def handler(context, event):
+      if event.path == "/do_something":
+        something()
+    ```
+  then now you should ensure the matching is against a byte-string, such as `b"/do_something"`.
```

Comment:
```suggestion
  In the new 3.7 and 3.8 runtimes, the matching `event.path` property is now a `byte-string` instead of an old `string`,
  The new snippet will look like this:
  <insert full snippet>
```